NewTrialBench.java

```diff
@@ -1,12 +1,35 @@
 import java.util.concurrent.ConcurrentHashMap;
 
 public class NewTrialBench {
-	static ConcurrentHashMap<Integer, String> h = new ConcurrentHashMap<Integer, String>();
+
+	private static ConcurrentHashMap<Integer, String> h = new ConcurrentHashMap<Integer, String>();
+
+  public String get(Integer key) {
+          if(h.containsKey(key)) {
+                  return h.get(key);
+          }
+          return "None";
+  }
+
+
+	public void put(Integer key, String value) {
+		synchronized(h) {
+			String obj = h.get(key);
+			if(obj == null) {
+				obj = value;
+				h.put(key,obj);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer



**Problem**  
 These lines use `ConcurrentHashMap`&#39;s **sequential** API and are protected by `synchronized` to ensure they execute together: **17 and 20**. This synchronization may reduce performance by limiting parallelism and does not take advantage of `ConcurrentHashMap`&#39;s **concurrent** API.

**Fix**  
 Consider replacing the check and `put()` with `putIfAbsent()` and then removing the synchronization. The check is at line: **18**.

**More info**  
 `putIfAbsent()` is thread-safe and puts the value only if the `ConcurrentHashMap` does not contain the key. `putIfAbsent()` returns `null` if the value did not exist and returns the value in the map if one already exists.

 [View an example on GitHub](https://github.com/apache/hadoop/blob/f16cf877e565084c66bc63605659b157c4394dc8/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-hs/src/main/java/org/apache/hadoop/mapreduce/v2/hs/HistoryFileManager.java#L927-L932) (external link).